core/src/xtdb/tx.clj

```diff
@@ -489,7 +502,32 @@
               (when (compare-and-set! !error nil t)
                 (when-not (instance? InterruptedException t)
                   (log/fatal t "Ingester error occurred"))
-                (bus/send bus {::xt/event-type ::ingester-error, :ingester-error t})))]
+                (bus/send bus {::xt/event-type ::ingester-error, :ingester-error t})))
+
+            ;; Applies a default transaction (fetch/encode) batching strategy that attempts to balance throughput and working set size
+            ;; (this will change if we can push this policy into each doc store)
+            ;; see docstring on tx-batch-preferred-doc-count
+            (batch-transactions [txs]
+              (lazy-seq
+                ((fn batch-txes [{:keys [batch, doc-hashes]
+                                  :or {batch [],
+                                       doc-hashes #{}}
+                                  :as buf} txs]
```

Comment:
style nit: try not to mix single-line params and multi-line params on the same line - it makes it harder to spot that there are in fact 2 params to this function